### ISSUE TITLE
an enhancement to prevent the code freeze

### ISSUE THREAD

**Issue by mdaliyan** [raff/godet#34]
I suggest you change this line
```
_ = <-done
```
to this:
```
	start := time.Now()
	var pageFound bool = false
DONE:
	for time.Since(start) < time.Second*2 {
		time.Sleep(time.Millisecond * 200)
		select {
		case pageFound = <-done:
			break DONE
		default:
		}
	}
	if ! pageFound {
		return errors.New("Page Not found")
	}
```
in case the domain does not exist, the first code will freezes the code.

**Comment by erikdubbelboer** [comment]
If you want to do that then at least use some proper code:
```go
timeout := time.After(time.Second*2)
select {
	case <-timeout:
		return errors.New("Page Not found")
	case <-done:
}
```

**Comment by mdaliyan** [comment]
No. The `done` channel will be set by `Page.frameStoppedLoading` event. it's not just a simple wait for 2 seconds.
```
	remote.CallbackEvent("Page.frameStoppedLoading", func(params godet.Params) { 
		done <- true
	})
```

**Comment by erikdubbelboer** [comment]
I'm not sure what you mean but my code and your code are 100% the same?

**Comment by raff** [comment]
Can you please specify where this enhancement should be made ?

**Comment by mdaliyan** [comment]
the problem was in one of the examples in this repo, that only has a blocking chanel `_ <- donè that caused freezing when I tried to reach an invalid code.

Your code is good @erikdubbelboer. actually way better than mine. I tested it when I reached home. my mistake. sorry.

**in brief:** this
```
_ = <-done
```
would be better to be replaced by this, in example files.
``` 
select {
	case <- time.After(time.Second*2):
		return errors.New("Page Not found")
	case <- done:
}
```
